heron/api/src/java/com/twitter/heron/streamlet/impl/streamlets/ConsumerStreamlet.java

```diff
@@ -38,13 +39,7 @@ public ConsumerStreamlet(StreamletImpl<R> parent, SerializableConsumer<R> consum
 
   @Override
   public boolean doBuild(TopologyBuilder bldr, Set<String> stageNames) {
-    if (getName() == null) {
-      setName(defaultNameCalculator("consumer", stageNames));
-    }
-    if (stageNames.contains(getName())) {
-      throw new RuntimeException("Duplicate Names");
-    }
-    stageNames.add(getName());
```

Comment:
Hi Luc. I was also thinking about this changes. Thanks for it. Adding streamlet name to stagesNames logic looks missed for `ConsumerStreamlet`. Other Streamlets look ok. I think this logic can also be part of the abstracted `setDefaultNameIfNone` method after streamlet name is uniqueness verification.